debezium-core/src/main/java/io/debezium/connector/common/BaseSourceConnector.java

```diff
@@ -0,0 +1,26 @@
+/*
+ * Copyright Debezium Authors.
+ *
+ * Licensed under the Apache Software License version 2.0, available at http://www.apache.org/licenses/LICENSE-2.0
+ */
+package io.debezium.connector.common;
+
+import java.util.List;
+import java.util.Map;
+
+import org.apache.kafka.common.config.ConfigValue;
+import org.apache.kafka.connect.source.SourceConnector;
+
+import io.debezium.config.Configuration;
+import io.debezium.rest.model.DataCollection;
+
+/**
+ * Base class for Debezium's CDC {@link SourceConnector} implementations.
+ * Provides functionality common to all CDC connectors, such as validation.
+ */
+public abstract class BaseSourceConnector extends SourceConnector {
+
+    protected abstract Map<String, ConfigValue> validateAllFields(Configuration config);
+
+    public abstract List<DataCollection> getMatchingCollections(Configuration config);
```

Comment:
@jpechane  I also don't think we should use a response model from the REST extension here `List<DataCollection>`? Would TableId or something else match better the "internal connector" use-case while for the UI we then need to translate that into the DataCollection model?

Reply:
@rk3rn3r Well, you can't as MongoDB does not use `TableId`. What you can do it to have generic parameter `T extends DataCollectionId`.

Reply:
I was not saying it must be `TableId`, just a suggestion (and I think those are new methods and if we decide that the method return `TableId` we could use a meaningful mapping to the TableId fields, this is required anyway). We could also create a value object as a copy from the `DataCollection` REST resonse model but I don't think it's a good idea to use the same entity for the connectors and for the REST response, although they are related there might be reasons in the future to diverge on the connector side implementation but we like to keep the REST response model unchanged.